pybossa/jobs.py

```diff
@@ -446,14 +447,37 @@ def import_tasks(project_id, **form_data):
     return msg
 
 
-def webhook(url, payload=None):
+def webhook(url, payload=None, oid=None):
     """Post to a webhook."""
-    import json
-    headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
-    if url:
-        return requests.post(url, data=json.dumps(payload), headers=headers)
-    else:
-        return False
+    from flask import current_app
+    try:
+        import json
+        from pybossa.core import sentinel, webhook_repo
+        headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+        if oid:
+            webhook = webhook_repo.get(oid)
+        else:
+            webhook = Webhook(project_id=payload['project_id'],
+                              payload=payload)
+        if url:
+            response = requests.post(url, data=json.dumps(payload), headers=headers)
+            webhook.response = response.text
+            webhook.response_status_code = response.status_code
+        else:
+            raise requests.exceptions.ConnectionError('Not URL')
```

Comment:
This exception is always catched in line 473, so why don't we save the try-except?